src/main/java/net/snowflake/ingest/streaming/internal/ParquetTypeGenerator.java

```diff
@@ -0,0 +1,302 @@
+package net.snowflake.ingest.streaming.internal;
+
+import java.util.*;
+import net.snowflake.ingest.utils.ErrorCode;
+import net.snowflake.ingest.utils.SFException;
+import org.apache.parquet.schema.LogicalTypeAnnotation;
+import org.apache.parquet.schema.PrimitiveType;
+import org.apache.parquet.schema.Type;
+import org.apache.parquet.schema.Types;
+
+/** Generates the Parquet types for the Snowflake's column types */
+public class ParquetTypeGenerator {
+
+  /**
+   * Util class that contains Parquet type and other metadata for that type needed by the Snowflake
+   * server side scanner
+   */
+  static class ParquetTypeInfo {
+    private Type parquetType;
+    private Map<String, String> metadata;
+
+    public Type getParquetType() {
+      return this.parquetType;
+    }
+
+    public Map<String, String> getMetadata() {
+      return this.metadata;
+    }
+
+    public void setParquetType(Type parquetType) {
+      this.parquetType = parquetType;
+    }
+
+    public void setMetadata(Map<String, String> metadata) {
+      this.metadata = metadata;
+    }
+  }
+
+  private static final Set<AbstractRowBuffer.ColumnPhysicalType> TIME_SUPPORTED_PHYSICAL_TYPES =
+      new HashSet<>(
+          Arrays.asList(
+              AbstractRowBuffer.ColumnPhysicalType.SB4, AbstractRowBuffer.ColumnPhysicalType.SB8));
+  private static final Set<AbstractRowBuffer.ColumnPhysicalType>
+      TIMESTAMP_SUPPORTED_PHYSICAL_TYPES =
+          new HashSet<>(
+              Arrays.asList(
+                  AbstractRowBuffer.ColumnPhysicalType.SB8,
+                  AbstractRowBuffer.ColumnPhysicalType.SB16));
+
+  /**
+   * Generate the column parquet type and metadata from the column metadata received from server
+   * side.
+   *
+   * @param column column metadata as received from server side
+   * @param id column id
+   * @return column parquet type
+   */
+  static ParquetTypeInfo generateColumnParquetTypeInfo(ColumnMetadata column, int id) {
+    ParquetTypeInfo res = new ParquetTypeInfo();
+    Type parquetType;
+    Map<String, String> metadata = new HashMap<>();
+    String name = column.getName();
+
+    AbstractRowBuffer.ColumnPhysicalType physicalType;
+    AbstractRowBuffer.ColumnLogicalType logicalType;
+    try {
+      physicalType = AbstractRowBuffer.ColumnPhysicalType.valueOf(column.getPhysicalType());
+      logicalType = AbstractRowBuffer.ColumnLogicalType.valueOf(column.getLogicalType());
+    } catch (IllegalArgumentException e) {
+      throw new SFException(
+          ErrorCode.UNKNOWN_DATA_TYPE, column.getLogicalType(), column.getPhysicalType());
+    }
+
+    metadata.put(Integer.toString(id), logicalType.getOrdinal() + "," + physicalType.getOrdinal());
+
+    // Parquet Type.Repetition in general supports repeated values for the same row column, like a
+    // list of values.
+    // This generator uses only
```

Comment:
This two lines seems short, is this done by the auto formatter?

Reply:
no, but change to one line.